spyder/config/user.py

```diff
@@ -447,14 +584,376 @@ def set(self, section, option, value, verbose=False, save=True):
             value = int(value)
         elif not is_text_string(default_value):
             value = repr(value)
+
         self._set(section, option, value, verbose)
         if save:
             self._save()
 
     def remove_section(self, section):
-        cp.ConfigParser.remove_section(self, section)
+        """Remove `section` and all options within it."""
+        super(UserConfig, self).remove_section(section)
         self._save()
 
     def remove_option(self, section, option):
-        cp.ConfigParser.remove_option(self, section, option)
+        """Remove `option` from `section`."""
+        super(UserConfig, self).remove_option(section, option)
         self._save()
+
+    def cleanup(self):
+        """Remove .ini file associated to config."""
+        os.remove(self.get_config_fpath())
+
+
+class SpyderUserConfig(UserConfig):
+
+    def get_previous_config_fpath(self):
+        """
+        Override method.
+
+        Return the last configuration file used if found.
+        """
+        fpath = self.get_config_fpath()
+        previous_paths = [
+            # >= 52.0.0
+            fpath,
+            # < 52.0.0
+            os.path.join(get_conf_path(), 'spyder.ini'),
+            # < ??
+        ]
+        for fpath in previous_paths:
+            print('fpath', fpath)
+            if osp.isfile(fpath):
+                break
+
+        return fpath
+
+    def get_config_fpath_from_version(self, version=None):
+        """
+        Override method.
+
+        Return the configuration path for given version.
+
+        If no version is provided, it returns the current file path.
+        """
+        fpath = self.get_config_fpath()
+
+        if version is None:
+            fpath = self.get_config_fpath()
+        elif check_version(version, '52.0.0', '<'):
+            fpath = osp.join(get_conf_path(), 'spyder.ini')
+        elif check_version(version, '2.4.0', '<'):
+            # TODO: Should this still be supported?
+            pass

```

Comment:
No, there's no need to support this anymore.

Reply:
Ok